block/produce.go

```diff
@@ -89,22 +89,14 @@ func (m *Manager) ProduceAndGossipBlock(ctx context.Context, allowEmpty bool) (*
 
 func (m *Manager) produceBlock(allowEmpty bool) (*types.Block, *types.Commit, error) {
 	var (
-		lastCommit     *types.Commit
-		lastHeaderHash [32]byte
-		newHeight      uint64
 		err            error
+		lastHeaderHash [32]byte
+		lastCommit     = &types.Commit{}
+		newHeight      = m.State.NextHeight()
 	)
 
-	if m.LastState.IsGenesis() {
-		newHeight = uint64(m.LastState.InitialHeight)
-		lastCommit = &types.Commit{}
-		m.LastState.BaseHeight = newHeight
-		if ok := m.Store.SetBase(newHeight); !ok {
-			return nil, nil, fmt.Errorf("store set base: %d", newHeight)
-		}
-	} else {
-		height := m.Store.Height()
-		newHeight = height + 1
+	if !m.State.IsGenesis() {
```

Comment:
as this `if` will be entered basically 100% of the times thinking if it would be cleaner to have a `produceGenesisBlock` and have part of the `produceBlock` code reusable.

Reply:
refactored it a bit, take a look